operators/pkg/controller/kibana/securesettings/initcontainer.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package securesettings
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/volume"
+	kbvolume "github.com/elastic/cloud-on-k8s/operators/pkg/controller/kibana/volume"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	InitContainerName = "init-keystore"
+)
+
+// script is a small bash script to create a Kibana keystore,
+// then add all entries from the secure settings secret volume into it.
+const script = `#!/usr/bin/env bash -eu
+
+echo "Initializing Kibana keystore."
+
+# create a keystore in the default data path
+./bin/kibana-keystore create
+
+# add all existing secret entries into it
+for filename in ` + kbvolume.SecureSettingsVolumeMountPath + `/*; do
+	[[ -e "$filename" ]] || continue # glob does not match
+	key=$(basename "$filename")
+	echo "Adding "$key" to the keystore."
+	./bin/kibana-keystore add "$key" --stdin < "$filename"
+done
+
+echo "Keystore initialization successful."
```

Comment:
We could do that, yes. I have mixed feelings though: that's another configmap to manage (a single one for multiple clusters?) + another volume to mount in the ES pod + another VolumeMount to setup in the init container to avoid 10 lines of bash in the pod spec (but probably adds more than 10 lines in the json-prettified pod spec because of additional volumes).
I can also reduce this to ~5 lines of bash.